@Test void shouldProcessAckFailureMessageWithPendingError() throws Exception {
  Neo4jError error=Neo4jError.from(new RuntimeException());
  connectionState.markFailed(error);
  assertEquals(error,connectionState.getPendingError());
  BoltStateMachineState newState=state.process(AckFailureMessage.INSTANCE,context);
  assertEquals(readyState,newState);
  assertNull(connectionState.getPendingError());
}
